### PR TITLE
A little rework

### PR DESCRIPTION
**Pull request by imaji**
Updated to .NET 4.6.2,
updated Nuget packages to latest.
Added in some basic tests, more to follow.
Removed reliance on events between WebSocket & Pusher/Channel objects

### CODE REVIEW

File: PusherClient/ConnectionState.cs [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
+        /// <summary>
+        /// The state after the Connection has been initialized
+        /// </summary>
         Initialized,
+        /// <summary>
+        /// The state when the connection process has begun
+        /// </summary>
         Connecting,
+        /// <summary>
+        /// The state when the connection process has cimpleted successfully
```
Comment (ewmy) [inline PusherClient/ConnectionState.cs]: completed :-)

File: PusherClient/ConnectionState.cs [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+        /// <summary>
+        /// The state when the disconnection process has begun
+        /// </summary>
+        Disconnecting,
+        /// <summary>
+        /// The state when the disconnection process has completed, or the connection was dropped
+        /// </summary>
         Disconnected,
+        /// <summary>
+        /// The state when a connectionr etry is in process
```
Comment (ewmy) [inline PusherClient/ConnectionState.cs]: connection retry

File: PusherClient/ConnectionStateChangedEventHandler.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿namespace PusherClient
+{
+    /// <summary>
+    /// The Event Handler for the Connection State Channged Event on the <see cref="Pusher"/>
```
Comment (ewmy) [inline PusherClient/ConnectionStateChangedEventHandler.cs]: Is there an "object" missing in this comment? Also, should be "changed"

**Comment by imaji** [comment]
Superseded

**Pull request closed without merging**